CONTRIBUTING.md

```diff
@@ -0,0 +1,95 @@
+
+When contributing to Iroh, you are expected to follow our [Code of Conduct][coc].
+
+Here are some of the ways in which you can contribute:
+
+## Discussions
+
+If you want to ask a question to understand a concept regarding Iroh, or need help working with Iroh, please check the [Discussions][discussions]. If you don't find a thread that fits your needs, feel free to create a new one. 
+
+## Issues
+
+If you found unexpected behavior while using Iroh, please browse our existing [issues][issues]. If no issues fit your case, [create a new one][newissue].
+
+If you would like to suggest a new feature in Iroh, [create a new issue][newissue]. This helps have meaningful conversations about design, feasibility, and general expectations of how a feature would work. If you plan to work on this yourself, we ask you to state this as well, so that you receive the guidance you need.
+
+## Pull requests
+
+Code contributions to Iroh are greatly appreciated. Here is the general workflow you should follow:
+
+1. **State in the associated issue your desire to work on it**
+  
+   If there is no issue for the work you would like to do, please open one. This helps reduce duplicated efforts and give contributors the help and guidance they might need.
+
+2. **Write some code!**
+
+   If this is your first contribution to Iroh, you will need to [fork][forkiroh] and clone it using git. If you need help with the code you are working on, don't hesitate to ask questions in the associated issue. We will be happy to help you.
+
+3. **Open the pull request**
+
+   In general, pull requests should be opened as [a draft][draftprs]. This way, the team and community can know what work is being done, and reviewers can give early pointers on the work you are doing. Additionally we ask you to follow these guidelines:
+
+   - **General code guidelines**
+
+     - When possible, please document relevant pieces of code. If documentation refers to another object (module, function, struct, etc) use ``[`path::to::ReferencedObject`]`` to link it. For more information check the [rustdoc docummentation][rustdoc]
+     - Comment your code. It will be useful for your reviewer and future contributors.
+
+   - **Pull request titles**
+ 
+     - Iroh pull requests titles look like this: `type(crate): description`
+
+       | **`type`** | **When to use** |
+       |--:         |-- |
+       | `feat`     | A new feature |
+       | `test`     | Changes that exclusively affect tests, either by adding new ones or correcting existing ones |
+       | `fix`      | A bug fix |
+       | `docs`     | Documentation only changes |
+       | `refactor` | A code change that neither fixes a bug nor adds a feature |
+       | `perf`     | A code change that improves performance |
+       | `deps`     | Dependency only updates |
+       | `chore`    | Changes to the build process or auxiliary tools and libraries |
+
+  
+       **`crate`** is the rust crate containing your changes.
```

Comment:
I'm not sure if this is the crate, I don't think I've put the crate there before.  The beetle guide describes this as "scope" which is probably better in that it is less strict.

Also, this was optional I think.  Sometimes I just don't know what to write and leave it out.

Reply:
it should be optional if you touch multiple pieces, now that we have multiple crates again the crate name will make the most sense in a lot of cases

Reply:
this was my thinking too. Right now point to a crate is the easiest way to go